### PR TITLE
create model and controller genrator

### LINKED ISSUE #9: Create model and controller generator.
- provide a generator rails g neeto_tags:<entity>, here entity can be anything which the user can provide. For example rails g neeto_tags:ticket would generate a ticket model inheriting from neeto tags with prefilled associations and add it in the parent application.
- rails g neeto_tags:ticket would also generate a api/v1/tickets/tags controller which would inherit from the tags controller of the engine. It would set your context to ticket tags example
`def tags
  @_tags ||= @organization.ticket_tags
end`

### PR DESCRIPTION
**Pull request by mayanksharma4132**
closes #9

### CODE REVIEW

**Comment by mayanksharma4132** [comment]
@chiraggshah _a
Please check.

File: lib/templates/models/entity.rb.tt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class <%= @class_name %> < Neeto::Tags::Tag
+    has_many :<%= @entity.downcase.pluralize %>,
```
Comment (chiraggshah) [inline lib/templates/models/entity.rb.tt:2]: Fix indentation in this file

File: test/dummy/app/controllers/ticket/tags_controller.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class Api::V1::Ticket::TagsController < Neeto::Tags::TagsController
+    private
```
Comment (chiraggshah) [inline test/dummy/app/controllers/ticket/tags_controller.rb:2]: Fix indentation in this file

File: test/dummy/app/models/ticket.rb [hunk 3]
```diff
@@ -1,6 +1,5 @@
 # frozen_string_literal: true
 
 class Ticket < ApplicationRecord
-  include Neeto::Tags::Taggable
-  tag_class_name('TicketTag')
+	include Neeto::Tags::Taggable
```
Comment (chiraggshah) [inline test/dummy/app/models/ticket.rb:4]: Fix indentation

File: test/dummy/app/models/ticket_tag.rb [hunk 4]
```diff
@@ -1,7 +1,5 @@
-# frozen_string_literal: true
-
 class TicketTag < Neeto::Tags::Tag
-  has_many :tickets,
-           through: :taggings, source: :taggable,
-           source_type: 'Ticket', class_name: 'Ticket'
+    has_many :tickets,
```
Comment (chiraggshah) [inline test/dummy/app/models/ticket_tag.rb:2]: Fix indentation

File: test/dummy/config/routes.rb [hunk 5]
```diff
@@ -2,4 +2,7 @@
 
 Rails.application.routes.draw do
   mount Neeto::Tags::Engine => '/neeto-tags'
+	namespace :ticket do
```
Comment (chiraggshah) [inline test/dummy/config/routes.rb:5]: Fix indentation

**Review by chiraggshah — COMMENTED** [review]
@mayanksharma4132 _a : Aded a few comments.

File: lib/generators/neeto_tags/tag_generator.rb [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+      argument :entity
+      source_root File.expand_path('../../templates', __dir__)
+
+      def create_entity_model
+        @entity = entity.camelize
+        @class_name = "#{@entity}Tag"
+        if Dir["app/models/#{@class_name.underscore}.rb"].any?
+          puts "#{@class_name.underscore}.rb has already been copied to your app"
+        else
+          template 'models/entity.rb', Rails.root + "app/models/#{@class_name.underscore}.rb"
```
Comment (chiraggshah) [inline lib/generators/neeto_tags/tag_generator.rb:14]: If I run `rails g neeto_tags:tag deal`, it creates a file `Deal.rb` whereas it should be `deal.rb`

**Review by chiraggshah — COMMENTED** [review]
At all the places use two spaces instead of `\t`

**Comment by mayanksharma4132** [comment]
@chiraggshah _a
Please check.

**Pull request merged**